src/FormMigration/Steps/FeeRecovery.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace Give\FormMigration\Steps;
+
+use Give\FormMigration\Contracts\FormMigrationStep;
+use Give\Framework\Blocks\BlockModel;
+
+class FeeRecovery extends FormMigrationStep
+{
+
+    /**
+     * @unreleased
+     */
+    public function process()
+    {
+        $feeRecoveryBlock = $this->fieldBlocks->findByName('givewp-fee-recovery/fee-recovery');
+        $feeRecoverySettings = $this->formV2->getFeeRecoverySettings();
+
+        if (empty($feeRecoverySettings)) {
+            if ($feeRecoveryBlock) {
+                $this->fieldBlocks->remove($feeRecoveryBlock->name);
+            }
+
+            return;
+        }
+
```

Comment:
I'm not sure what's going on here. The form migration should just be concerned with v2 to v3.  I don't think we need to be removing blocks.

Reply:
The Fee Recovery block is automatically included in new forms when the global setting is enabled. If we are migrating a v2 form where Fee Recovery has been disabled, we should remove the block from the form. This is necessary because the new form is added before the migration steps are executed.

Reply:
Since the form is being built programmatically, the migration should be bypassing the `givewp_form_builder_new_form` action that default options are hooking into.  If that's _not_ the case then we need to update the migration processing because the migration should not be concerned with those side effects for brand new forms.

Reply:
In fact, changed the migration on e5f8ca134265bc06d4a0fb2d9d4ebb98bdaaa4b1 and it doesn't include the block automatically.